content/blog/llm-agentic-tool-mesh-empowering-gen-ai-with-retrieval-augmented-generation-rag.md

```diff
@@ -0,0 +1,267 @@
+
+Example usage
+
+```python
+# Configuration for the Data Extractor
+EXTRACTOR_CONFIG = {
+    'type': 'UnstructuredSections',
+    'document_type': 'Pdf',
+    'cache_elements_to_file': True,
+    'extract_text': True,
+    'exclude_header': True,
+    'exclude_footer': True,
+    'extract_image': False,
+    'image_output_folder': './images'
+}
+
+# Initialize the Data Extractor
+data_extractor = DataExtractor.create(EXTRACTOR_CONFIG)
+
+# Parse a document file
+file_path = 'example_document.pdf'
+result = data_extractor.parse(file_path)
+
+# Handle the extraction result
+if result.status == "success":
+    print(f"EXTRACTED ELEMENTS:\n{result.elements}")
+else:
+    print(f"ERROR:\n{result.error_message}")
+```
+
+### Transformation
+
+As part of the transformation phase, the process involves cleaning the data by removing irrelevant or redundant information, enriching it with metadata to enhance searchability during retrieval, and transforming the cleaned data using LLMs to generate summaries, question-and-answer pairs, or other structured outputs.
+
+Example usage
+
+````python
+from athon.rag import DataTransformer
+
+# Configuration for the Data Transformer
+TRANSFORMER_CONFIG = {
+    'type': 'CteActionRunner',
+    'clean': {
+        'headers_to_remove': ['Confidential', 'Draft'],
+        'min_section_length': 100
+    },
+    'transform': {
+        'llm_config': {
+            'type': 'LangChainChatOpenAI',
+            'api_key': 'your-api-key-here',
+            'model_name': 'gpt-4o'
+        },
+        'system_prompt': 'Summarize the following content.',
+        'transform_delimeters': ['```', '```json']
+    },
+    'enrich': {
+        'metadata': {
+            'source': 'LLM Agentic Tool Mesh Platform',
+            'processed_by': 'CteActionRunner'
+        }
+    }
+}
+
+# Initialize the Data Transformer
+data_transformer = DataTransformer.create(TRANSFORMER_CONFIG)
+
+# List of extracted elements to be transformed
+extracted_elements = [
+    {"text": "Confidential Report on AI Development", "metadata": {"type": "Header"}},
+    {"text": "AI is transforming industries worldwide...", "metadata": {"type": "Paragraph"}}
+]
+
+# Define the actions to be performed
+actions = ['RemoveSectionsByHeader', 'TransformInSummary', 'EnrichMetadata']
+
+# Process the elements
+result = data_transformer.process(actions, extracted_elements)
+
+# Handle the transformation result
+if result.status == "success":
+    print(f"TRANSFORMED ELEMENTS:\n{result.elements}")
+else:
+    print(f"ERROR:\n{result.error_message}")
+````
+
+### Loading
+
+As part of the loading phase, the process includes injecting the transformed data into the chosen storage solution, such as a vector database, and further adapting the data as needed, such as by chunking it into smaller pieces for efficient retrieval.
```

Comment:
Change "such as by chunking it..." into "such as chunking it..." (take out the word "by")